nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -515,13 +940,20 @@ Nav2Panel::onNewGoal(double x, double y, double theta, QString frame)
   pose.pose.position.z = 0.0;
   pose.pose.orientation = orientationAroundZAxis(theta);
 
-  if (state_machine_.configuration().contains(accumulating_)) {
-    acummulated_poses_.push_back(pose);
+  if (store_poses_.empty()) {
```

Comment:
Why the check on store poses here?

Reply:
Is there a less fragile way to check what you need to do here?

Reply:
This is kind of essential. We do not want to have any accidentally selected waypoints making it's way into the waypoint list. 

I'm bit unsure why you say it fragile?

Reply:
Because we go the pause state and return back to the accumulated state.. In between, we should make sure that we don't update the pose array with the mistakenly given goals from the RViz..

Reply:
I thought you may have been using that as a proxy for the state, this is fine